org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java

```diff
@@ -303,14 +304,14 @@ public static void assertDiagnostics(List<Diagnostic> actual, Diagnostic... expe
 	public static void assertDiagnostics(List<Diagnostic> actual, List<Diagnostic> expected, boolean filter) {
 		List<Diagnostic> received = actual;
 		final boolean filterMessage;
-		if(expected != null && !expected.isEmpty() && expected.get(0).getMessage() != null) {
+		if(expected != null && !expected.isEmpty() && expected.get(0).getMessage() != null && expected.get(0).getMessage().isEmpty() == false) {

```

Comment:
better to use concise code like 

!expected.get(0).getMessage().isEmpty()

Reply:
`if(expected != null && !expected.isEmpty() && !StringUtils.isEmpty(expected.get(0).getMessage()))`